knowledge_repo/app/routes/auth.py

```diff
@@ -0,0 +1,56 @@
+from flask import flash, redirect, render_template, request, current_app, Blueprint, url_for
+from flask_login import login_user, logout_user, login_required
+from flask_principal import identity_changed, AnonymousIdentity, PermissionDenied
+
+from ..models import User
+from ..proxies import current_app
+from ..utils.auth import is_safe_url
+
+blueprint = Blueprint('auth', __name__,
+                      template_folder='../templates', static_folder='../static')
+
+
+@blueprint.route('/auth/login', methods=['GET', 'POST'])
+def login():
+
+    providers = current_app.auth_providers
+
+    if len(providers) == 1:
+        return redirect(url_for('auth_provider.{}.prompt'.format(providers[0].name)))
+
+    return render_template(
+        'auth-switcher.html',
+        providers=[{'name': provider.name, 'icon_uri': provider.icon_uri, 'link_text': provider.link_text} for provider in providers]
+    )
+
+# @blueprint.route('/auth/login/<provider>', methods=['GET', 'POST'])
```

Comment:
Yep. In my haste to be done with the patch, I left some unused sample code in there. Thanks for picking it out.